pkg/client/config/loader.go

```diff
@@ -0,0 +1,55 @@
+package config
+
+import (
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+const (
+	OpenShiftConfigPathEnvVar      = "KUBECONFIG"
+	OpenShiftConfigFlagName        = "config"
+	OpenShiftConfigHomeDir         = ".kube"
+	OpenShiftConfigHomeFileName    = "config"
+	OpenShiftConfigHomeDirFileName = OpenShiftConfigHomeDir + "/" + OpenShiftConfigHomeFileName
+)
+
+var RecommendedHomeFile = path.Join(homedir.HomeDir(), OpenShiftConfigHomeDirFileName)
+
+// currentMigrationRules returns a map that holds the history of recommended home directories used in previous versions.
+// Any future changes to RecommendedHomeFile and related are expected to add a migration rule here, in order to make
+// sure existing config files are migrated to their new locations properly.
+func CurrentMigrationRules() map[string]string {
```

Comment:
do the commands using this care about migration rules?  Which ones are they?

Reply:
Our wrapper command for `NewCmdConfig` [calls cmdconfig.NewOpenShiftClientConfigLoadingRules](https://github.com/openshift/origin/blob/master/pkg/oc/cli/cmd/wrappers.go#L745), which in turn [calls `currentMigrationRules`](https://github.com/openshift/origin/blob/master/pkg/oc/cli/config/loader.go#L45)